### DISCUSSION TITLE
Can React 18 be used with Next.js version 15?

### DISCUSSION THREAD

**Discussion by toshi-ino** [vercel/next.js#72795] [App Router]
Dear All and Next.js Team,

I have a question about Next.js version 15.

In my current project, I would like to use Next.js version 15. According to the documentation, it says to use React 19. However, other packages used in the project cannot be used with React 19.

When I look at Next.js's `package.json`, it is written as follows:

node_modules/next/package.json
```
    "react": "^18.2.0 || 19.0.0-rc-66855b96-20241106",
    "react-dom": "^18.2.0 || 19.0.0-rc-66855b96-20241106",

```

I understand this to mean that Next.js 15 also works with React 18; is this correct?

We don't have much time. It would be helpful if we could get quick feedback.

Thank you.

**Comment by jacksteves** [comment]
You can downgrade for now, yep. Check out https://ui.shadcn.com/docs/react-19

**Reply by toshi-ino** [reply]
Thank you for your response! It looks like documentation for shadcn/ui. Am I correct in understanding that it will work with React 18 in a Next 15+ App Router setup? My understanding is that shadcn/ui is compatible with React 18 as well.

**Comment by toshi-ino** [comment]
It seems that my previous question lacked some details. I would like to know if Next 15 (App Router) + React 18 will work. My current understanding is that Next 15 (Page Router) + React 18 works, but Next 15 (App Router) requires React 19.

**Comment by aquibbaig** [comment]
I think next15's app router requires React 19 RC, but you can use the pages router with React 18. Check here: https://nextjs.org/blog/next-15#react-19

**Comment by riyatechie12** [comment]
Yes, you're correct. Next.js 15 supports both React 18 and the React 19 release candidate.
